src/stringobj.c

```diff
@@ -1757,32 +1731,30 @@ Obj FuncTranslateString (
   Int j, len;
 
   /* check whether <string> is a string                                  */
-  if ( ! IsStringConv( string ) ) {
+  while ( ! IsStringConv( string ) ) {
     string = ErrorReturnObj(
-	     "RemoveCharacters: first argument <string> must be a string (not a %s)",
+	     "TranslateString: first argument <string> must be a string (not a %s)",
 	     (Int)TNAM_OBJ(string), 0L,
 	     "you can replace <string> via 'return <string>;'" );
-    return FuncTranslateString( self, string, trans );
   }
   
-  /* check whether <trans> is a string                                  */
-  if ( ! IsStringConv( trans ) ) {
-    trans = ErrorReturnObj(
-	     "RemoveCharacters: second argument <trans> must be a string (not a %s)",
-	     (Int)TNAM_OBJ(trans), 0L,
-	     "you can replace <trans> via 'return <trans>;'" );
-    return FuncTranslateString( self, string, trans );
-  }
- 
-  /* check if string has length at least 256 */
-  if ( GET_LEN_STRING( trans ) < 256 ) {
-    trans = ErrorReturnObj(
-	     "RemoveCharacters: second argument <trans> must have length >= 256",
-	     0L, 0L,
-	     "you can replace <trans> via 'return <trans>;'" );
-    return FuncTranslateString( self, string, trans );
+  // check whether <trans> is a string of length at least 256
+  while ( ! IsStringConv( trans ) || GET_LEN_STRING( trans ) < 256 ) {
+    if ( ! IsStringConv( trans ) ) {
+      trans = ErrorReturnObj(
+           "TranslateString: second argument <trans> must be a string (not a %s)",
+           (Int)TNAM_OBJ(trans), 0L,
+           "you can replace <trans> via 'return <trans>;'" );
+    }
+
+    if ( GET_LEN_STRING( trans ) < 256 ) {
```

Comment:
Oops, this must be changed to ˋelse ifˋ (user might have returned e.g. a boolean)